### ISSUE TITLE
VnV probes: Train Data

### ISSUE THREAD

**Issue by MarcBehrens** [openETCS/validation#288]
Following information gives a hint to where to look for probes within the model.
Related Specification: SUBSET-027 v 3.1.0
Concerning: Train Data
JRU NID_MESSAGE: 2

SUBSET-027 Variable | SCADE Variable| Declaration |  filename
---|---|---|---
V_MAXTRAIN | cICE3TrainData.maxTrainSpeed |TIU_Types_Pkg::trainData_T | openETCS_EVC.xscade

**Comment by T12z** [comment]
@MarcBehrens which constants are you looking for?
Related to **M_BRAKE_GAMMA_CONF**, these do not exist yet and may not exist for an intermediate period. I just now started modelling the conversion model which is based on the brake percentage. In the end both will be implemented, it is just a matter of time.

Though those kind of SRS-constants have the history of being defined outside of SpeedSupervision and we just import them. This is due to the fact that inputs to our package are already of the specific ie. enumeration type.

On a larger, probably unrelated sidenote:
most of our constants are defined within SDM_Types::. Some helper constants do float around in the sub projects, but I may consolidate this at some stage. We have also converted most (all?) default-constants for array-types into "/dev/zero"-factory-operators. So if array sizes are adapted for any reason we do not have to touch constants.

**Comment by MarcBehrens** [comment]
@t12z basically I am looking to where the following veriables are defined/ can be probed

SUBSET-027 Variable |	SCADE Variable |	Declaration |	filename
---|---|---|---		
Q_BRAKE_CAPT_TYPE	|||		
M_BRAKE_PERCENTAGE	|||		
N_BRAKE_CONF	|||		
M_BRAKE_LAMBDA_CONF(k)	|||		
T_BRAKE_SERVICE(k)	|||		
N_BRAKE_CONF	|||		
M_BRAKE_GAMMA_CONF(k)	|||		
T_BRAKE_EMERGENCY(k)	|||		
N_BRAKE_SECTIONS(k)	|||		
V_BRAKE_EMERGENCY_COMP(k, m)	|||		
A_BRAKE_EMERGENCY_COMP(k, m)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 0)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 1)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 2)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 3)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 4)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 5)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 6)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 7)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 8)	|||		
M_KDRY_RST(A_BRAKE_EMERGENCY_COMP(k, m), 9)	|||		
M_KWET_RST(A_BRAKE_EMERGENCY_COMP(k, m))	|||		
T_BRAKE_SERVICE(k)	|||		
N_BRAKE_SECTIONS(k)	|||		
V_BRAKE_SERVICE_COMP(k, m)	|||		
A_BRAKE_SERVICE_COMP(k, m)	|||

**Comment by T12z** [comment]
brief: not yet implemented, (though I am at it currently).

When you see the scheme SUBSET-026-3 v330 3.13.1.3 / Figure 28, we started from the right and have finished blocks *commands*, *limits*, *determination of curves + targets*, *MRSP*. *decelerations* is partially done. *build up times*, *gradient*, *models*, *conversion* are up next and are currently faked by constants to have a working model.
Above variables may find its way into upcomming "SDM_Models" package. Though I am still in the process of digesting the SRS, so the architecture is not yet fixed.

Depending on the amount of other "noise" I receive, I may have something finished by the end of the month.

**Comment by MarcBehrens** [comment]
T_TRACTION_CUT_OFF currently blocked by openETCS/modeling#542

**Comment by MarcBehrens** [comment]
@T12z did you already implement the missing variables?
Could you please give an update on the fixed assumptions.

**Comment by T12z** [comment]
I need to have a closer look tomorrow.
I do not see a clear mapping for a few of them, and on the other hand, I will have to post pone work on the gamma-train related things. It will still take time to get the conversion model (lambda train) running in the net weeks.
Sorry.

**Comment by T12z** [comment]
Soooooo,

SUBSET-027 Variable | comment
--- | ---
T_TRACTION_CUT_OFF | no progress
M_BRAKE_GAMMA_CONF | gamma,  postponed to later release
Q_BRAKE_CAPT_TYPE | ?
M_BRAKE_PERCENTAGE | ok, TIU
N_BRAKE_CONF | ?, prob. gamma related
M_BRAKE_LAMBDA_CONF(k) | "conf"?, staging
T_BRAKE_SERVICE(k) | k=...?, staging
N_BRAKE_CONF | ?? like above?
M_BRAKE_GAMMA_CONF(k) | gamma
T_BRAKE_EMERGENCY(k) | like T_BRAKE_SERVICE
N_BRAKE_SECTIONS(k) | ?, prob. gamma related
V_BRAKE_EMERGENCY_COMP(k, m) | ?
A_BRAKE_EMERGENCY_COMP(k, m) | ? 
M_KDRY_RST(*) | gamma
M_KWET_RST(*) | gamma		
T_BRAKE_SERVICE(k) | above??
N_BRAKE_SECTIONS(k) | above??
V_BRAKE_SERVICE_COMP(k, m) | ?
A_BRAKE_SERVICE_COMP(k, m) | ?

as you see, quite a few question marks, looking forward to the call :+1:
